### PR TITLE
riak-ts 1.3.0 (new formula)

### PR DESCRIPTION
**Pull request by rrooding**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/riak-ts.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class RiakTs < Formula
+  desc "NoSQL time series database optimized for IoT and Time Series data."
+  homepage "http://basho.com/products/riak-ts/"
+  url "http://s3.amazonaws.com/downloads.basho.com/riak_ts/1.3/1.3.0/osx/10.8/riak-ts-1.3.0-OSX-x86_64.tar.gz"
```
Comment (zmwangx) [inline Formula/riak-ts.rb]: You're downloading a binary distribution, which is not allowed. Need to install from source.

File: Formula/riak-ts.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  version "1.3.0"
```
Comment (zmwangx) [inline Formula/riak-ts.rb]: Redundant, remove this line.

File: Formula/riak-ts.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      c.gsub! /(platform_log_dir *=).*$/, "\\1 #{logdir}"
+    end
+    bin.write_exec_script libexec/"bin/riak"
+    bin.write_exec_script libexec/"bin/riak-admin"
+    bin.write_exec_script libexec/"bin/riak-debug"
+    bin.write_exec_script libexec/"bin/riak-shell"
+    bin.write_exec_script libexec/"bin/search-cmd"
+  end
+
+  test do
```
Comment (zmwangx) [inline Formula/riak-ts.rb]: Need to write a test.

**Review by zmwangx — CHANGES_REQUESTED** [review]

File: Formula/riak-ts.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on :macos => :mountain_lion
+  depends_on :arch => :x86_64
+
+  def install
+    logdir = var + "log/riak-ts"
+    datadir = var + "lib/riak-ts"
+    libexec.install Dir["*"]
+    logdir.mkpath
+    datadir.mkpath
+    (datadir + "ring").mkpath
```
Comment (bfontaine) [inline Formula/riak-ts.rb]: You can replace this and the line above it with `(datadir/"ring").mkpath`.

File: Formula/riak-ts.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    inreplace "#{libexec}/lib/env.sh" do |s|
+      s.change_make_var! "RUNNER_BASE_DIR", libexec
+      s.change_make_var! "RUNNER_LOG_DIR", logdir
```
Comment (bfontaine) [inline Formula/riak-ts.rb]: Can’t we override these in the environment?

File: Formula/riak-ts.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    end
+    inreplace "#{libexec}/etc/riak.conf" do |c|
+      c.gsub! /(platform_data_dir *=).*$/, "\\1 #{datadir}"
+      c.gsub! /(platform_log_dir *=).*$/, "\\1 #{logdir}"
+    end
+    bin.write_exec_script libexec/"bin/riak"
+    bin.write_exec_script libexec/"bin/riak-admin"
+    bin.write_exec_script libexec/"bin/riak-debug"
+    bin.write_exec_script libexec/"bin/riak-shell"
+    bin.write_exec_script libexec/"bin/search-cmd"
```
Comment (bfontaine) [inline Formula/riak-ts.rb]: You can factorize this:

```rb
%w[riak riak-admin riak-debug riak-shell search-cmd].each do |script|
  bin.write_exec_script libexec/"bin/#{script}"
end
```

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/riak-ts.rb (hunk 6 above)
Comment (zmwangx) [inline Formula/riak-ts.rb]: I'd hold off requesting changes in `install` before the building from source issue is addressed.

**Comment by rrooding** [comment]
Thanks for the comments, I'll take another look at it tomorrow!

**Comment by woodruffw** [comment]
Closing due to inactivity.

**Pull request closed without merging**